### DISCUSSION TITLE
Infer header name from `accessorKey` or `id`

### DISCUSSION THREAD

**Discussion by githorse** [KevinVandy/material-react-table#525] [Ideas]
99% of the time the header can be derived from the `accessorKey` via some very simple capitalization / snake- and camel-case parsing rules:

`age` -> `Age`
`firstName` -> "First Name"
`first_name` -> "First Name"

It would be great to be able to infer that. ([AG Grid](https://www.ag-grid.com/) does this.)

Of course, people might differ in their capitalization preferences ("First name" vs "First Name", for instance), so it would be super cool to pass in an optional function telling the grid how to derive the header: `(accessorKeyOrId: string) => string`.

**Comment by KevinVandy** [comment]
just type 1 more property. I don't want to mess around with localization stuff here. Sometimes explicitness is better than saving lines of code.
